### ISSUE TITLE
Trouble using rwr-node-dev-server

### ISSUE THREAD

**Issue by alduethadyn** [netguru/react_webpack_rails#138]
When attempting to start the node server, and run the examples, I'm running into a number of issues:

Running `npm start` (webpack reports it was successful) in a terminal and `rails s` in the other and then navigating to http://localhost:3000/react_examples/component fails with: 

```
ActionView::Template::Error (Failed to open TCP connection to localhost:8081 (Connection refused - connect(2) for "localhost" port 8081)):
    1: <%= react_component('HelloWorld', { name: @name }, { server_side: true }) %>

app/views/react_examples/component.html.erb:1:in `_app_views_react_examples_component_html_erb__1535323047093588030_70187131793780'
  Rendering /Users/abeckwith/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/templates/rescues/template_error.html.erb within rescues/layout
  Rendering /Users/abeckwith/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/templates/rescues/_source.html.erb
  Rendered /Users/abeckwith/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/templates/rescues/_source.html.erb (4.4ms)
  Rendering /Users/abeckwith/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/templates/rescues/_trace.html.erb
  Rendered /Users/abeckwith/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/templates/rescues/_trace.html.erb (2.0ms)
  Rendering /Users/abeckwith/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/templates/rescues/_request_and_response.html.erb
  Rendered /Users/abeckwith/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/templates/rescues/_request_and_response.html.erb (0.8ms)
  Rendered /Users/abeckwith/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/actionpack-5.0.0.1/lib/action_dispatch/middleware/templates/rescues/template_error.html.erb within rescues/layout (86.2ms)
```

I'm guessing the above is expected since the example uses server-side rendering, so I tried `npm rwr-node-dev-server` but this yields:
```
> @ rwr-node-dev-server /Users/abeckwith/Code/scratch/origin/origin_meals_erp
> forever forever/development.json

warn:    --minUptime not set. Defaulting to: 1000ms
warn:    --spinSleepTime not set. Your script will exit if it does not stay up for at least 1000ms
[Function: HttpDispatcher]
/Users/abeckwith/Code/scratch/origin/origin_meals_erp/app/react/node_server.js:32
dispatcher.onPost('/run', (req, res) => {
           ^

TypeError: dispatcher.onPost is not a function
    at Object.<anonymous> (/Users/abeckwith/Code/scratch/origin/origin_meals_erp/app/react/node_server.js:32:12)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.runMain (module.js:590:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
```

(`[Function: HttpDispatcher]` is from the console.log I added to node_server.js - `console.log(dispatcher.onPost)` yields `undefined`)

**Versions in ruby-land:**
Rails 5.0.0.1
Ruby 2.3.0
react_webpack_rails 0.6.0

**Versions in node-land:**
node v6.5.0
react ^15.1.0
react-webpack-rails ^0.5.0

I dropped a console.log before this line to inspect the dispatcher object, which does seem to be a function, but all of the prototype/functions added in httpdispatcher.js are seemingly missing. The version in node_modules seems to contain the right bits. 

Let me know if you want access to my repo - as it's pretty vanilla at this point, but is currently private.

Any ideas on where I'm going wrong here? It seems like require can't find the modules, but then again, the Function HttpDispatcher seems present, so I'm a little baffled.

**Comment by alduethadyn** [comment]
npm warns:
`npm WARN deprecated httpdispatcher@1.1.0: This version is not retrocompatible, use 2.0.0 instead`

**Comment by alduethadyn** [comment]
Changing the generated node_server.js require'd-in httpdispatcher bits to:
```
const dispatcher = new (require('httpdispatcher'))();
```
...seems to get things working - still makes me feel like I've set something up incorrectly?

**Comment by mxaly** [comment]
hello @alduethadyn and sory for late response.

httpdispatcher is a constructor in new version so you need to call `new` on it. 
it will be fixed in v0.7 generator.  
for now you can stay with your fix or use updated nodeserver from here: https://github.com/netguru/react_webpack_rails/blob/release_v0.7.0/lib/generators/react_webpack_rails/templates/react/node_server.js
